phpseclib/Crypt/AES.php

```diff
@@ -67,31 +67,32 @@ class AES extends Rijndael
      *
      * @see \phpseclib\Crypt\Rijndael::setBlockLength()
      * @access public
-     * @param int $length
+     * @throws \BadMethodCallException anytime it's called
      */
     function setBlockLength($length)
     {
-        return;
+        throw new \BadMethodCallException('The block length cannot be set for AES.');
     }
 
     /**
      * Sets the key length
      *
-     * Valid key lengths are 128, 192, and 256.  If the length is less than 128, it will be rounded up to
-     * 128.  If the length is greater than 128 and invalid, it will be rounded down to the closest valid amount.
+     * Valid key lengths are 128, 192, and 256.  Set the link to bool(false) to disable a fixed key length
      *
      * @see \phpseclib\Crypt\Rijndael:setKeyLength()
      * @access public
      * @param int $length
+     * @throws \LengthException if the key length isn't supported
      */
     function setKeyLength($length)
     {
         switch ($length) {
-            case 160:
-                $length = 192;
+            case 128:
+            case 192:
+            case 256:
                 break;
-            case 224:
-                $length = 256;
+            default:
+                throw new \LengthException('Key of size ' . strlen($key) . ' not supported by this algorithm. Only keys of sizes 16, 24 or 32 supported');
```

Comment:
* The `$key` variable doesn't exist in this function. Should it be `$length`?
* Should the following be rather "Key **length** of"? (For consistency with the function name.)

  > Key of **size**

* Should the following be rather "Only key **lengths of 128, 192 or 256** supported."? (For consistency with the function name, and in bits.)

  > Only keys of **sizes 16, 24 or 32** supported

Note: This could be a copy-paste error from https://github.com/phpseclib/phpseclib/pull/942/files#diff-05acee93a75378713d649bc1b9601be2R119.